### PR TITLE
More robust EA-MUS detection/parsing

### PR DESCRIPTION
**Pull request by ghost**
Recreating this PR as I think I've added the needed bounds checking now, but this should help load EA-MUS songs whose first byte is not 0x7D.

### CODE REVIEW

File: include/midi_sequencer_impl.hpp [hunk 1]
```diff
@@ -2250,12 +2250,12 @@ static bool detectRSXX(const char *head, FileAndMemReader &fr)
     bool ret = false;
 
     // Try to identify RSXX format
-    if(head[0] == 0x7D)
+    if (head[0] >= 0x5D && fr.fileSize() > head[0])
```
Comment (Wohlstand) [inline include/midi_sequencer_impl.hpp:2253]: The Code style of this project needs to don't have the space between `if` and the bracket. (1)

File: include/midi_sequencer_impl.hpp [hunk 2]
```diff
@@ -2250,12 +2250,12 @@ static bool detectRSXX(const char *head, FileAndMemReader &fr)
...
     {
-        fr.seek(0x6D, FileAndMemReader::SET);
+        fr.seek(head[0] - 0x10, FileAndMemReader::SET);
         fr.read(headerBuf, 1, 6);
         if(std::memcmp(headerBuf, "rsxx}u", 6) == 0)
-            ret = true;
+                ret = true;
```
Comment (Wohlstand) [inline include/midi_sequencer_impl.hpp:2258]: Wrong indenting of this line, there are 4 spaces indenting used here.

File: include/midi_sequencer_impl.hpp [hunk 3]
```diff
@@ -2496,14 +2496,20 @@ bool BW_MidiSequencer::parseRSXX(FileAndMemReader &fr)
     }
 
     // Try to identify RSXX format
-    if(headerBuf[0] == 0x7D)
+    char start = headerBuf[0];
+    if (start < 0x5D)
```
Comment (Wohlstand) [inline include/midi_sequencer_impl.hpp:2500]: Same as (1).

**Review by Wohlstand — COMMENTED** [review]
It seems fine, but I suggest you fix a code style.

Note: when you contribute to any project, ensure you keep the same code style as the project that you contribute.

**Comment by Wohlstand** [comment]
Okay, I fixed these style cases by myself.

**Comment by ghost** [comment]
Sorry about that; I had ported over some changes that I had made when implementing your sequencer in my own project so I didn't think about matching the style. I'll keep that in mind if I make future PRs. I appreciate you handling it in the meantime, though.

**Comment by Wohlstand** [comment]
By the way, I took your update into other projects where this thing is used, and ye, with your changes, there are other songs that started to work (I tested "Ski or Die" and "The Immortal", these packs were available quickly). However, as I know, they have other banks, but I used Cartooners' bank to play them, and they worked, but with not exactly the same instruments. If you could help me to rip these banks, I'll be happy to include them in my libADLMIDI.
- Cartooners' bank is placed here: https://github.com/Wohlstand/libADLMIDI/tree/master/fm_banks/misc_files
- The code that parses it into the embedded database: https://github.com/Wohlstand/libADLMIDI/blob/master/utils/gen_adldata/file_formats/load_ea.h

I am not sure, are these banks the same, or do they have different structures?

**Pull request merged**